py_dtn7/__init__.py

```diff
@@ -2,5 +2,5 @@
 
 from .bundle import Bundle  # noqa F401
 from .dtn_rest_client import DTNRESTClient  # noqa F401
-from .dtn_ws_client import DTNWSClient, WSMode  # noqa F401
+# from .dtn_ws_client import DTNWSClient, WSMode  # noqa F401
```

Comment:
Would you consider importing this module conditionally like you are already doing it with `requests`? The use-case for the WebSocket interface is slim, but it exists.
```suggestion
try:
    from .dtn_ws_client import DTNWSClient, WSMode  # noqa F401
except ImportError:
    # Websocket implementation not installed or not supported on platform
    pass
```

Another possible way to do this would be to explicitly check the name of the Python implementation. In case of an `ImportError` the import would then still correctly fail in a misconfigured environment:

```suggestion
import sys
if sys.implementation.name != 'micropython':
    from .dtn_ws_client import DTNWSClient, WSMode  # noqa F401
```
This also applies to the other conditional imports.

Reply:
Actually, this was a mistake on my side because, as you rightly assumed, the websocket implementation cannot be used/imported on Micropython.

For minimal changes I would suggest the first approach, because the try-except approach is also used in the other modules. Maybe a comment on the other try-excepts would be useful too.

The cleanest way (an my preferred way) is probably to do the second approach on each CPython/Micropython specific import. (cbor2/cbor, requests/urequests)
If you want to do this, then you could also consider to define a constant like "RUNNING_MICROPYTHON" in utils.py

Reply:
Do you mean defining the constant like this:
```python
RUNNING_MICROPYTHON = sys.implementation.name == "micropython"
```
and then checking `RUNNING_MICROPYTHON` before the imports? That would provide much better readability, yes.

Reply:
Yes, exactly. Only downside is that the imports must be ordered correctly. But, that can also be something positive.